includes/base_controls/QControlBase.class.php

```diff
@@ -257,11 +257,16 @@ public function __construct($objParentObject, $strControlId = null) {
 			if (strlen($strControlId) == 0)
 				$this->strControlId = $this->objForm->GenerateControlId();
 			else {
+				if (substr($strControlId, 0, 6) === 'Qform_') {
```

Comment:
`ctype_alnum` already ensures that no underscore symbol would be allowed. This check is unnecessary.